spec/lib/api/v3/work_packages/work_packages_shared_helpers_spec.rb

```diff
@@ -0,0 +1,89 @@
+#-- encoding: UTF-8
+#-- copyright
+# OpenProject is a project management system.
+# Copyright (C) 2012-2015 the OpenProject Foundation (OPF)
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See doc/COPYRIGHT.rdoc for more details.
+
+require 'spec_helper'
+
+describe ::API::V3::WorkPackages::WorkPackagesSharedHelpers do
+  include ::API::V3::WorkPackages::WorkPackagesSharedHelpers
+
+  # Literally mocking the environment for the test here
+  # This is necessary since a module cannot be instantiated
+  let(:env) { { 'api.request.body' => { subject: 'foo' } } }
+
+  let(:work_package) { FactoryGirl.create(:work_package) }
+  let(:current_user) { FactoryGirl.build(:admin) }
+
+  before do
+    allow(self).to receive(:status).with(an_instance_of(Fixnum))
+  end
```

Comment:
I'd probably move this before statement up to the `let(:env)`, because it is part of mocking the environment. As opposed to `work_package` and `current_user`, which are part of the spec...

OMG!
`current_user` is also a valid method in the context where the helpers are mounted...

Probably you should rather do something like:

````
let(:helped_class) {
  Class.new do
    include ::API::V3::WorkPackages::WorkPackagesSharedHelpers

    def env
      { ... } # I believe you could even return stuff from the spec here, but not sure...
    end

    def status(code)
    end
  end
}

let(:instance_of_that_thing) { helped_class.new }
````

The thing I'd like to avoid is that by adding a `let` to the test, the whole inclusion breaks apart, because the let defines/overwrites a method of the helper...